raitools/raitools/_managers/error_analysis_manager.py

```diff
@@ -3,10 +3,58 @@
 
 """Defines the Error Analysis Manager class."""
 
-from raitools._internal.constants import ManagerNames
+from pathlib import Path
+import json
+from raitools._internal.constants import (
+    ManagerNames, ListProperties, ErrorAnalysisManagerKeys as Keys)
 from raitools._managers.base_manager import BaseManager
 from raitools._config.base_config import BaseConfig
 from erroranalysis._internal.error_analyzer import ModelAnalyzer
+from erroranalysis._internal.error_report import (
+    json_converter as report_json_converter, as_error_report)
+
+REPORTS = 'reports'
+CONFIG = 'config'
+MAX_DEPTH = 'max_depth'
+NUM_LEAVES = 'num_leaves'
+FILTER_FEATURES = 'filter_features'
+
+
+def config_json_converter(obj):
+    """Helper function to convert ErrorAnalysisConfig object to json.
+
+    :param obj: Object to convert to json.
+    :type obj: object
+    :return: The converted json.
+    :rtype: dict
+    """
+    if isinstance(obj, ErrorAnalysisConfig):
+        rdict = obj.__dict__
+        return rdict
+    try:
+        return obj.to_json()
+    except AttributeError:
+        return obj.__dict__
+
+
+def as_error_config(json_dict):
+    """Helper function to convert json to an ErrorAnalysisConfig object.
+
+    :param json_dict: The json to convert.
+    :type json_dict: dict
+    :return: The converted ErrorAnalysisConfig.
+    :rtype: ErrorAnalysisConfig
+    """
+    has_max_depth = MAX_DEPTH in json_dict
+    has_num_leaves = NUM_LEAVES in json_dict
+    has_filter_features = FILTER_FEATURES in json_dict
+    if has_max_depth and has_num_leaves and has_filter_features:
```

Comment:
This is how I'd do it since I don't mind using more than 80 col, but your way is totally fine. Feel free to close this comment.

```python
has_max_depth = MAX_DEPTH in json_dict
has_num_leaves = NUM_LEAVES in json_dict
has_filter_features = FILTER_FEATURES in json_dict
if has_max_depth and has_num_leaves and has_filter_features:
    max_depth = json_dict[MAX_DEPTH]
    num_leaves = json_dict[NUM_LEAVES]
    filter_features = json_dict[FILTER_FEATURES]
    return ErrorAnalysisConfig(max_depth, num_leaves, filter_features)

# vs.

if all(attribute in json_dict for attribute in [MAX_DEPTH, NUM_LEAVES, FILTER_FEATURES]):
    return ErrorAnalysisConfig(json_dict[MAX_DEPTH], json_dict[NUM_LEAVES], json_dict[FILTER_FEATURES])
```

Reply:
oh no, we need to follow 80 col limit for flake8